warehouse/query-core/src/main/java/datawave/query/jexl/lookups/BoundedRangeIndexLookup.java

```diff
@@ -137,20 +134,17 @@ public synchronized void submit() {
                 bs.setRanges(Collections.singleton(range));
                 bs.fetchColumnFamily(new Text(literalRange.getFieldName()));
 
-                // set up the GlobalIndexRangeSamplingIterator
-
-                IteratorSetting cfg = new IteratorSetting(config.getBaseIteratorPriority() + 50, "WholeRowIterator", WholeRowIterator.class);
-                bs.addScanIterator(cfg);
-
-                cfg = new IteratorSetting(config.getBaseIteratorPriority() + 48, "DateFilter", ColumnQualifierRangeIterator.class);
-                // search from 20YYddMM to 20ZZddMM\uffff to ensure we encompass all of the current day
-                String end = endDay + Constants.MAX_UNICODE_STRING;
-                cfg.addOption(ColumnQualifierRangeIterator.RANGE_NAME, ColumnQualifierRangeIterator.encodeRange(new Range(startDay, end)));
-
-                bs.addScanIterator(cfg);
+                IteratorSetting setting = new IteratorSetting(config.getBaseIteratorPriority() + 20, "BoundedRangeExpansionIterator",
+                                BoundedRangeExpansionIterator.class);
+                setting.addOption(BoundedRangeExpansionIterator.START_DATE, startDay);
+                setting.addOption(BoundedRangeExpansionIterator.END_DATE, endDay);
+                if (!config.getDatatypeFilter().isEmpty()) {
+                    setting.addOption(BoundedRangeExpansionIterator.DATATYPES_OPT, Joiner.on(',').join(config.getDatatypeFilter()));
+                }
+                bs.addScanIterator(setting);
 
                 // If this is a composite field, with multiple terms, we need to setup our query to filter based on each component of the composite range
-                if (config.getCompositeToFieldMap().get(literalRange.getFieldName()) != null) {
+                if (!config.getCompositeToFieldMap().get(literalRange.getFieldName()).isEmpty()) {
```

Comment:
Do we need to check for null as well?

Reply:
We do not. The `Multimap.get()` call returns an empty collection when nothing maps to the provided key. Which means that yes, we were slapping this iterator onto the stack every single time for no reason.